presto-spi/src/main/java/com/facebook/presto/spi/type/FloatType.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static io.airlift.slice.SizeOf.SIZE_OF_FLOAT;
+
+public final class FloatType
+        extends AbstractFixedWidthType
+{
+    public static final FloatType FLOAT = new FloatType();
+
+    private FloatType()
+    {
+        super(parseTypeSignature(StandardTypes.FLOAT), long.class, SIZE_OF_FLOAT);
+    }
+
+    @Override
+    public boolean isComparable()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean isOrderable()
+    {
+        return true;
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+        return block.getFloat(position, 0);
+    }
+
+    @Override
+    public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        int leftValue = leftBlock.getInt(leftPosition, 0);
+        int rightValue = rightBlock.getInt(rightPosition, 0);
+        return leftValue == rightValue;
+    }
+
+    @Override
+    public long hash(Block block, int position)
+    {
+        return (long) Float.floatToRawIntBits(block.getFloat(position, 0));
+    }
+
+    @Override
+    public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        // WARNING: the correctness of InCodeGenerator is dependent on the implementation of this
```

Comment:
@raghavsethi since I followed the same pattern of storing `float` (32b) within `long` as you did with `integer` type because there's no writer for `int` yet, I assume this comment is still valid here, isn't it?